File: client/src/lib/invoke/getLogin.ts
import { invoke } from "@tauri-apps/api/core";
import { ResultAsync } from "neverthrow";
import type { TokenType } from "@bindings";

export function getLogin(
    gitlab_token_type: TokenType,
    gitlab_token: string
): ResultAsync<string, string> {
    return ResultAsync.fromPromise(
        invoke("get_login", { gitlab_token_type, gitlab_token }),
        (error) => `Failed to get login: ${error}`
    );
}

File: client/src-tauri/src/commands/get_login.rs
use nexium::gitlab::{GitlabClient, TokenType};

#[tauri::command]
pub async fn get_login(
    gitlab_token_type: TokenType,
    gitlab_token: String,
) -> Result<String, String> {
    tauri::async_runtime::spawn_blocking(move || {
        GitlabClient::new(gitlab_token, gitlab_token_type)
            .get_login()
            .map_err(|err| err.to_string())
    })
    .await
    .map_err(|err| format!("Failed to get login: {}", err))?
}
